File: project-service/src/main/java/com/kapitonau/projectservice/model/ProjectModel.java
package com.kapitonau.projectservice.model;

import com.kapitonau.projectservice.model.audit.BaseAuditableModel;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "projects")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(BaseAuditableModel.class)
public class ProjectModel extends BaseAuditableModel {

    @Id
    @Column(name = "project_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "projects_project_id_seq")
    @SequenceGenerator(name = "projects_project_id_seq", sequenceName = "projects_project_id_seq", allocationSize = 1)
    private Long projectId;

    @Column(name = "space_id")
    private Long spaceId;

    @Column(name = "title")
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "project_status_id")
    private Long projectStatusId;

    @Column(name = "created_date")
    private LocalDate createdDate;

    @Column(name = "expected_close_date")
    private LocalDate expectedCloseDate;

}

File: workspace-service/src/main/kotlin/com/kapitonau/ps/workspaceservice/model/ProjectModel.kt
package com.kapitonau.ps.workspaceservice.model

import com.kapitonau.ps.workspaceservice.model.audit.BaseAuditModel
import com.kapitonau.ps.workspaceservice.model.audit.BaseAuditModelListener
import jakarta.persistence.*

@Entity
@Table(name = "projects")
@EntityListeners(BaseAuditModelListener::class)
open class ProjectModel: BaseAuditModel() {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "projects_id_gen")
    @SequenceGenerator(name = "projects_id_gen", sequenceName = "projects_project_id_seq", allocationSize = 1)
    @Column(name = "project_id", nullable = false)
    open var projectId: Long? = null

    @Column(name = "workspace_id", nullable = false)
    open var workspaceId: Long? = null

    @Column(name = "project_title", nullable = false, length = 256)
    open var projectTitle: String? = null

    @Column(name = "description", nullable = false, length = Integer.MAX_VALUE)
    open var description: String? = null

    @Column(name = "project_type_id", nullable = false)
    open var projectTypeId: Long? = null

    @Column(name = "is_deleted", nullable = false)
    open var isDeleted: Boolean? = null
}